File: Arrays101/283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if(len(nums) < 2): return 
        pointer = 0
        for x in nums:
            if x != 0:
                nums[pointer] = x
                pointer+=1
        while(pointer < len(nums)): 
            nums[pointer] = 0
            pointer+=1


        

File: Arrays101/283. Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        //check edge cases
        if(nums.length < 2) return;
        int pointer = 0;
        for(int x = 0; x<nums.length;x++){
            if(nums[x]!=0){
                nums[pointer] = nums[x];
                pointer++;
            }
        }
        while(pointer <nums.length){
            nums[pointer] = 0;
            pointer++;
        }
    }
}